web-app/src/components/BookingCalendar/container.js

```diff
@@ -36,21 +40,65 @@ const BookingCalendarContainer = Wrapped =>
       this.props.setEventBeingUpdated(isBeingUpdated);
     };
 
-    onSelectSlot = ({ start, end }) => {
+    checkIfPastDatesSelected = start => {
       const today = new moment();
-      if (moment(start).isAfter(today.add(-1, 'days'))) {
+      return moment(start).isBefore(today);
+    };
+
+    selectedDatesOverlapExisting = (start, end) => {
+      const overlappingEvents = this.props.takenHolidays.filter(hol => {
+        if (hol.employee && hol.employee.employeeId === this.props.employeeId) {
```

Comment:
Destructure these props

Reply:
I didn't destructure them as employee might not exists in the hol object. hol.employee need to exist first, which is the first check in the if statement.

Reply:
That shouldn't matter. You can still destructure, it's just if `hol.employee` isn't there then the destructured `employee` would be given the value of `undefined`. I had to double check this but:

![code_2018-08-23_13-44-50](https://user-images.githubusercontent.com/39301552/44526067-c3a12780-a6da-11e8-875d-34143bfd76ab.png)

Reply:
ok, updated and pushed up. Personality, I think its not needed and adds another line of code